public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx

```diff
@@ -109,10 +114,16 @@ export const LokiQueryBuilder = React.memo<Props>(
         setSampleData(sampleData);
       };
 
-      if (config.featureToggles.lokiQueryHints) {
+      const updateBasedOnChangedTimeRange =
```

Comment:
Prometheus does this slightly differently, instead of checking to see if the time range has changed within a certain threshold, it will round the start/end time to the nearest nth minute

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/language_utils.ts#L382

Your solution here should prevent double-querying opposed to the prom solution when a user fires a query at :59.69 and again at :00.00, so maybe this is a better way

Reply:
But maybe we want to pull this out into a helper that can be used in other places within loki?

Reply:
The advantage of the range snapping is that request durations are more uniform across users which could potentially lead to better caching in loki, but this fix can ship on its own before we look into that.